### PR TITLE
Add DataFrame.es_query() to modify Elasticsearch queries directly

### LINKED ISSUE #143: Support Elasticsearch queries
As the data is being pulled from Elasticsearch, there is a whole range of search functions available that could be exposed to allow filtering of the data retrieved.
For example, analysed text and fuzzy queries could be used to identify a corpus for analysis in eland.

We could support this through something like mirroring the elasticsearch-py search functions 
https://elasticsearch-py.readthedocs.io/en/master/api.html#elasticsearch.Elasticsearch.search
https://elasticsearch-py.readthedocs.io/en/master/api.html#elasticsearch.Elasticsearch.msearch

That way the user could choose between the pandas compatible/mirrored functions or the Elasticsearch specific extensions.

### PR DESCRIPTION
**Pull request by sethmlarson**
Follows the form of prefixing anything Elasticsearch-related with `es_`.

Let me know if you think this should be implemented differently, this seemed like the straightforward way to me and seemed to work for a handful of query types and combinations. :)

Allows users to specify queries like so:

```python
>>> import eland as ed
>>> ed.DataFrame("localhost", "ecommerce")
>>> df.es_query({
  "bool": {
    "filter": {
      "geo_distance": {
        "distance": "1km",
        "geoip.location": [55.3, 25.3]
      }
    }
  }
})[["category", "geoip.location"]].head(5)
                                                  category              geoip.location
1                                       [Women's Clothing]  {'lon': 55.3, 'lat': 25.3}
9                        [Women's Shoes, Women's Clothing]  {'lon': 55.3, 'lat': 25.3}
10  [Women's Accessories, Women's Clothing, Women's Shoes]  {'lon': 55.3, 'lat': 25.3}
22                       [Women's Shoes, Women's Clothing]  {'lon': 55.3, 'lat': 25.3}
30                                        [Men's Clothing]  {'lon': 55.3, 'lat': 25.3}
```

Closes #143.

### CODE REVIEW

File: docs/source/reference/api/eland.DataFrame.es_query.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+eland.DataFrame.es_query
```
Comment (stevedodson) [inline docs/source/reference/api/eland.DataFrame.es_query.rst:1]: pandas autogenerates these rst files, we should maybe look to do this longer term

File: eland/dataframe.py [hunk 2]
```diff
@@ -569,6 +569,54 @@ def info_es(self):
...
+
+        Examples
+        --------
+
+        Apply a `geo-distance query`_ to a dataset with a geo-point column ``geoip.location``.
+
+         .. _geo-distance query: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-geo-distance-query.html
+
+        >>> df = ed.DataFrame('localhost', 'ecommerce', columns=['customer_first_name', 'geoip.city_name'])
+        >>> df.es_query({"bool": {"filter": {"geo_distance": {"distance": "1km", "geoip.location": [25.3, 55.3]}}}})
```
Comment (stevedodson) [inline eland/dataframe.py:593]: Other examples include results which pass `pytest --doctest-modules` - or skip test. Probably good to review doctests.

File: docs/source/reference/api/eland.DataFrame.es_query.rst (hunk 1 above)
Comment (sethmlarson) [reply docs/source/reference/api/eland.DataFrame.es_query.rst:1]: Agreed! I'll create an issue for this.

Comment (sethmlarson) [reply docs/source/reference/api/eland.DataFrame.es_query.rst:1]: Created in: https://github.com/elastic/eland/issues/160

File: eland/dataframe.py (hunk 2 above)
Comment (sethmlarson) [reply eland/dataframe.py:593]: Ah that's how you run the doc tests! Should I add that to the `test` nox task?

Comment (sethmlarson) [reply eland/dataframe.py:593]: Doc tests now run as a part of CI.

**Comment by stevedodson** [comment]
LGTM!

**Pull request merged**